File: shared/pkg/enforce/session/task.go
// Package session provides session state management.
// task.go: Task state management functions.
// DACE: Single responsibility - task tracking only.
// Called by: gates/task.go (TaskCreate/TaskUpdate) and memory/sync.go (Write/Edit).
package session

// SetTask updates the current task being worked on.
// Called by: task gate on TaskUpdate with status change.
func (s *SessionState) SetTask(task, status string) {
	s.CurrentTask = task
	s.TaskStatus = status
	s.Save()
}

// AddFileModified tracks a file that was modified in this session.
// Returns true if the file was newly added, false if already tracked.
// Called by: memory sync on PostToolUse:Write/Edit.
func (s *SessionState) AddFileModified(filePath string) bool {
	for _, f := range s.FilesModified {
		if f == filePath {
			return false
		}
	}
	s.FilesModified = append(s.FilesModified, filePath)
	s.Save()
	return true
}

// ClearTask clears the current task state.
// Called by: task gate on task completion/deletion.
func (s *SessionState) ClearTask() {
	s.CurrentTask = ""
	s.TaskStatus = ""
	s.FilesModified = []string{}
	s.Save()
}

// HasTask returns true if a task is currently active.
// Called by: CEO gate to check if a task context exists.
func (s *SessionState) HasTask() bool {
	return s.CurrentTask != ""
}

File: crates/kavach-session/src/task.rs
// SOURCE: ~/.claude/CLAUDE.md Code form § nano-files: tests in separate mapped files
use crate::state::SessionState;

impl SessionState {
    pub fn set_task(&mut self, task: &str, status: &str) {
        self.current_task = task.into();
        self.task_status = status.into();
        self.save_or_log();
    }

    pub fn add_file_modified(&mut self, file_path: &str) -> bool {
        if self.files_modified.iter().any(|f| f == file_path) {
            return false;
        }
        self.files_modified.push(file_path.into());
        self.save_or_log();
        true
    }

    #[must_use]
    pub const fn has_task(&self) -> bool {
        !self.current_task.is_empty()
    }
}
